### PR TITLE
Add serverchan3 support for  serverchan notification provider

### PR DESCRIPTION
**Pull request by easychen**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #(issue)

UptimeKuma has a notification provider called ServerChan that released a new branch version. This PR adds compatibility for the new branch (by adjusting the API entry through the Sendkey recognition)

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

File: server/notification-providers/serverchan.js [hunk 1]
```diff
@@ -11,8 +11,12 @@ class ServerChan extends NotificationProvider {
     async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
         const okMsg = "Sent Successfully.";
 
+        const url = String(notification.serverChanSendKey).startsWith('sctp')
```
Comment (CommanderStorm) [inline server/notification-providers/serverchan.js:14]:
```suggestion
        // serverchan3 requires sending via ft07.com
        const url = String(notification.serverChanSendKey).startsWith("sctp")
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Given that below changes the URL, could you please link to the docs where this is recommended?
Seems a bit sus otherwise to change to a different second level domain..

**Comment by easychen** [comment]
Official Website: https://sc3.ft07.com
Official Announcement: [Chinese](https://mp.weixin.qq.com/s?__biz=MzIzMDA2MDMyNg==&mid=2648303923&idx=1&sn=5e8c29eebd392d97ee851066dbeafda9&chksm=f094717ac7e3f86cbec5419fa973cee196f860e4463949554fdc4e7732bb17a0c901d4c32403&token=1424278085&lang=zh_CN#rd )
I add a txt under sct.ftqq.com: https://sct.ftqq.com/md/sc3.txt

The reason for changing sendkey to be included in the domain is due to the constant cyclic requests to the API interface by some third-party programs. By using DNS to direct subdomains to 127.0.0.1, we can alleviate the burden on the server.😅

**Comment by CommanderStorm** [comment]
Thanks for the adjustments to the notification provider! 🎉 



> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>

**Pull request merged**